### PR TITLE
✨vesting: streams consolidation panel

### PR DESCRIPTION
**Pull request by franm91**
image

### CODE REVIEW

File: pages/vesting.tsx [hunk 1]
```diff
@@ -24,9 +49,16 @@ const Vesting: NextPage = () => {
   const { activeStreams, loading: streamsLoading, refetch } = useUpdateStreams();
   const { switchNetwork, isLoading: switchIsLoading } = useSwitchNetwork();
   const { data: reserveRatio } = useEscrowedEXAReserveRatio();
+  const { reservedData, reservedIsLoading, withdrawableAmount, withdrawableAmountIsLoading } = useEscrowEXATotals(
+    activeStreams.map(({ tokenId }) => Number(tokenId)),
+  );
 
   const [loading, setLoading] = useState<boolean>(false);
+  const [showAll, setShowAll] = useState<boolean>(false);
+  const [CancelModalOpen, setCancelModalOpen] = useState(false);
```
Comment (sebipap) [inline pages/vesting.tsx:58]:
```suggestion
  const [cancelModalOpen, setCancelModalOpen] = useState(false);
```

File: pages/vesting.tsx [hunk 2]
```diff
@@ -36,10 +68,43 @@ const Vesting: NextPage = () => {
...
+    );
+  }, [activeStreams]);
+
+  const totalWithdrawableAmount = useMemo(() => {
+    if (!withdrawableAmount) return 0n;
+
+    return withdrawableAmount.reduce((acc, { result }) => acc + (result as bigint), 0n) || 0n;
+  }, [withdrawableAmount]);
+
+  const allCancelabelStremsIds = useMemo(() => {
```
Comment (sebipap) [inline pages/vesting.tsx:97]: typo

File: pages/vesting.tsx [hunk 3]
```diff
@@ -66,6 +131,163 @@ const Vesting: NextPage = () => {
     }
   }, [activeStreams, escrowedEXA, opts, refetch]);
 
+  const { totalVestedEsEXA, totalWithdrawnEXA } = totalsStreamData;
+
+  const toogleTable = () => {
```
Comment (sebipap) [inline pages/vesting.tsx:136]:
```suggestion
  const toogleShowAll = () => {
```
for more clarity

File: pages/vesting.tsx [hunk 4]
```diff
@@ -66,6 +131,163 @@ const Vesting: NextPage = () => {
...
+    setShowAll(!showAll);
+  };
+
+  const closeCancelModal = useCallback(() => {
+    setCancelModalOpen(false);
+  }, []);
+
+  const handleWithdrawAll = useCallback(() => {
```
Comment (sebipap) [inline pages/vesting.tsx:144]:
```suggestion
  const withdrawAll = useCallback(() => {
```

**Review by sebipap — APPROVED** [review]
lgtm.
I add minor naming suggestions

File: pages/vesting.tsx [hunk 5]
```diff
@@ -66,6 +131,163 @@ const Vesting: NextPage = () => {
...
+  const Transition = React.forwardRef(function Transition(
+    props: TransitionProps & {
+      children: React.ReactElement;
+    },
+    ref: React.Ref<unknown>,
+  ) {
+    return <Slide direction="up" ref={ref} {...props} />;
+  });
+
+  const WithdrawAndCancel: React.FC<{
```
Comment (jgalat) [inline pages/vesting.tsx:195]: Maybe define these 3 outside? These are being re-declared on every re-render of `Vesting`

File: pages/vesting.tsx [hunk 6]
```diff
@@ -66,6 +131,163 @@ const Vesting: NextPage = () => {
     }
   }, [activeStreams, escrowedEXA, opts, refetch]);
 
+  const { totalVestedEsEXA, totalWithdrawnEXA } = totalsStreamData;
+
+  const toogleShowAll = () => {
```
Comment (jgalat) [inline pages/vesting.tsx:136]: typo, `toggle`

**Review by jgalat — APPROVED** [review]
lgtm, minor comments

**Review by sebipap — APPROVED** [review]

**Review by jgalat — DISMISSED** [review]

**Review by jgalat — APPROVED** [review]

**Review by sebipap — APPROVED** [review]

**Pull request merged**